integration/admin-client/src/main/java/org/keycloak/admin/client/Keycloak.java

```diff
@@ -41,10 +42,45 @@
  * @see KeycloakBuilder
  */
 public class Keycloak implements AutoCloseable {
+
+    private static volatile ResteasyClientProvider CLIENT_PROVIDER = resolveResteasyClientProvider();
+
+    private static ResteasyClientProvider resolveResteasyClientProvider() {
+        Iterator<ResteasyClientProvider> providers = ServiceLoader.load(ResteasyClientProvider.class).iterator();
+
+        if (providers.hasNext()) {
+            ResteasyClientProvider provider = providers.next();
+
+            if (providers.hasNext()) {
+                throw new IllegalArgumentException("Multiple " + ResteasyClientProvider.class + " implementations found");
+            }
```

Comment:
As ResteasyClientClassicProvider is on the same module as this class it will always be loaded won't it? So it doesn't seem like it would possible to use a different provider?